coderd/workspaces_test.go

```diff
@@ -3722,3 +3723,50 @@ func TestWorkspaceTimings(t *testing.T) {
 		})
 	}
 }
+
+func TestGetAuthorizedWorkspacesAndAgents(t *testing.T) {
+	t.Parallel()
+
+	ownerClient, _, api := coderdtest.NewWithAPI(t, &coderdtest.Options{
+		IncludeProvisionerDaemon: true,
+	})
+	owner := coderdtest.CreateFirstUser(t, ownerClient)
+	_, user := coderdtest.CreateAnotherUser(t, ownerClient, owner.OrganizationID)
+	authToken := uuid.NewString()
+	version := coderdtest.CreateTemplateVersion(t, ownerClient, owner.OrganizationID, &echo.Responses{
+		Parse:          echo.ParseComplete,
+		ProvisionPlan:  echo.PlanComplete,
+		ProvisionApply: echo.ProvisionApplyWithAgent(authToken),
+	})
+	coderdtest.AwaitTemplateVersionJobCompleted(t, ownerClient, version.ID)
+	template := coderdtest.CreateTemplate(t, ownerClient, owner.OrganizationID, version.ID)
+	workspace := coderdtest.CreateWorkspace(t, ownerClient, template.ID)
+	coderdtest.AwaitWorkspaceBuildJobCompleted(t, ownerClient, workspace.LatestBuild.ID)
+	_ = agenttest.New(t, ownerClient.URL, authToken)
+	_ = coderdtest.NewWorkspaceAgentWaiter(t, ownerClient, workspace.ID).Wait()
```

Comment:
Spinning up `coderdtest` for this test feels like overkill.  You should be able to use `dbgen` to populate the database with the rows you'll query.

Reply:
This test is just for checking the auth filter works, the test in `querier_test` is far more comprehensive, and uses dbgen to populate it.

Reply:
The auth filter is at the dbauthz layer, it doesn't need coderd.

Reply:
I'm not sure how else you'd test that the auth-filter string is being inserted into the query correctly at runtime.
An existing query using it has this comment:
```
s.Run("GetAuthorizedWorkspaces", s.Subtest(func(db database.Store, check *expects) {
  _ = dbgen.Workspace(s.T(), db, database.Workspace{})
  _ = dbgen.Workspace(s.T(), db, database.Workspace{})
  // No asserts here because SQLFilter.
  check.Args(database.GetWorkspacesParams{}, emptyPreparedAuthorized{}).Asserts()
}))
```

Reply:
Oh, duh - my bad, thanks!